nixos/modules/services/x11/desktop-managers/mate.nix

```diff
@@ -78,9 +75,6 @@ in
         # Add mate-control-center paths to some XDG variables because its schemas are needed by mate-settings-daemon, and mate-settings-daemon is a dependency for mate-control-center (that is, they are mutually recursive)
         ${addToXDGDirs pkgs.mate.mate-control-center}
 
-        # Update user dirs as described in http://freedesktop.org/wiki/Software/xdg-user-dirs/
-        ${pkgs.xdg-user-dirs}/bin/xdg-user-dirs-update
```

Comment:
Is this not needed anymore?

Reply:
Don't think so. even if the autostart doesn't have the key to select the autostart phase for mate.

Reply:
I see all user directories in a vm with this branch.

![screenshot from 2019-03-04 21 12 24](https://user-images.githubusercontent.com/28888242/53775946-32fa9e00-3eec-11e9-939e-30a79e126c9f.png)

Reply:
Hmm, checking out https://freedesktop.org/wiki/Software/xdg-user-dirs/ it seems that that update thing will also localize these directories and do some more stuff.

Reply:
Yep, I've done the same for plasma https://github.com/NixOS/nixpkgs/pull/56553

Reply:
Ahh, I didn't notice the `xdg-user-dirs` in `systemPackages`. But I don't really get how that even works, because I thought putting packages in `systemPackages` would just make them available in /run/current-system/sw.

Oh I bet mate automatically runs the binary if it finds it! If so I feel like a purer solution would be to patch mate for it not to depend on the system path for this.

Reply:
> Ahh, I didn't notice the xdg-user-dirs in systemPackages. But I don't really get how that even works, because I thought putting packages in systemPackages would just make them available in /run/current-system/sw.

GNOME and Mate have implemented the [`XDG Autostart Spec`](https://specifications.freedesktop.org/autostart-spec/autostart-spec-latest.html) and all desktop files placed at `XDG_CONFIG_DIRS/autostart` will be started by the [session manager](https://gitlab.gnome.org/GNOME/gnome-session/tree/master/gnome-session).

We do https://github.com/NixOS/nixpkgs/blob/8a8056c302bf395f70fa72ac52984e4b3698bb9a/nixos/modules/programs/environment.nix#L41 so it all works.

> Oh I bet mate automatically runs the binary if it finds it! If so I feel like a purer solution would be to patch mate for it not to depend on the system path for this.

Only thing I can see that could be better is wrapping xdg-user-dirs's autostarts. (like [this](https://github.com/NixOS/nixpkgs/blob/2fcb11a2442b27834db87b2f62511037c31b5e36/pkgs/desktops/gnome-3/core/gnome-session/default.nix#L50)).